include/zephyr/bluetooth/gatt.h

```diff
@@ -640,7 +640,7 @@ ssize_t bt_gatt_attr_read_service(struct bt_conn *conn,
  */
 #define BT_GATT_SECONDARY_SERVICE(_service)				\
 	BT_GATT_ATTRIBUTE(BT_UUID_GATT_SECONDARY, BT_GATT_PERM_READ,	\
-			 bt_gatt_attr_read_service, NULL, _service)
+			 bt_gatt_attr_read_service, NULL, (void *)_service)
```

Comment:
@Thalley 
> You haven't shared what the compiler warning is, which is what I am requesting :)

if not add `void *` there, compiler will warning for that:

<img width="1079" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/26378606/9365778d-16a2-40d8-9389-f3324b587863">

Reply:
But, only warning, not build error.